src/Polly.Extensions/Telemetry/Log.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Polly.Extensions.Telemetry;
+
+#pragma warning disable S107 // Methods should not have too many parameters
+#pragma warning disable SA1118 // Parameter should not span multiple lines
+
+internal static class Log
```

Comment:
Can we conditionally compile this class so we can use the Logging Source Generator for .NET 6+ for better performance?

Reply:
Done.